pkg/storage/swift/swift.go

```diff
@@ -1,84 +1,223 @@
-				SecretKeyRef: &corev1.SecretKeySelector{
-					LocalObjectReference: corev1.LocalObjectReference{
-						Name: imageregistryv1.ImageRegistryPrivateConfiguration,
-					},
-					Key: "REGISTRY_STORAGE_SWIFT_USERNAME",
-				},
-			},
-		},
-		corev1.EnvVar{
-			Name: "REGISTRY_STORAGE_SWIFT_PASSWORD",
-			ValueFrom: &corev1.EnvVarSource{
-				SecretKeyRef: &corev1.SecretKeySelector{
-					LocalObjectReference: corev1.LocalObjectReference{
-						Name: imageregistryv1.ImageRegistryPrivateConfiguration,
-					},
-					Key: "REGISTRY_STORAGE_SWIFT_PASSWORD",
-				},
-			},
-		},
-	)
+	var username, password string
 
-	if d.Config.Domain != "" {
-		envs = append(envs, corev1.EnvVar{Name: "REGISTRY_STORAGE_SWIFT_DOMAIN", Value: d.Config.Domain})
+	cfg, err := clusterconfig.GetSwiftConfig(d.Listers)
+	if err != nil {
+		return nil, err
 	}
 
-	if d.Config.DomainID != "" {
-		envs = append(envs, corev1.EnvVar{Name: "REGISTRY_STORAGE_SWIFT_DOMAINID", Value: d.Config.DomainID})
-	}
+	if cfg.Storage.Swift.Cloud != nil {
+		cloud := cfg.Storage.Swift.Cloud
 
-	if d.Config.Tenant != "" {
-		envs = append(envs, corev1.EnvVar{Name: "REGISTRY_STORAGE_SWIFT_TENANT", Value: d.Config.Tenant})
-	}
+		d.Config.AuthURL = replaceEmpty(d.Config.AuthURL, cloud.AuthInfo.AuthURL)
+		d.Config.Tenant = replaceEmpty(d.Config.Tenant, cloud.AuthInfo.ProjectName)
+		d.Config.TenantID = replaceEmpty(d.Config.TenantID, cloud.AuthInfo.ProjectID)
+		d.Config.Domain = replaceEmpty(d.Config.Domain, cloud.AuthInfo.DomainName)
+		d.Config.DomainID = replaceEmpty(d.Config.DomainID, cloud.AuthInfo.DomainID)
+		d.Config.Domain = replaceEmpty(d.Config.Domain, cloud.AuthInfo.UserDomainName)
+		d.Config.DomainID = replaceEmpty(d.Config.DomainID, cloud.AuthInfo.UserDomainID)
+		d.Config.RegionName = replaceEmpty(d.Config.RegionName, cloud.RegionName)
 
-	if d.Config.TenantID != "" {
-		envs = append(envs, corev1.EnvVar{Name: "REGISTRY_STORAGE_SWIFT_TENANTID", Value: d.Config.TenantID})
+		username = cloud.AuthInfo.Username
+		password = cloud.AuthInfo.Password
+	} else {
+		username = cfg.Storage.Swift.Username
+		password = cfg.Storage.Swift.Password
 	}
+
+	envs = append(envs,
+		corev1.EnvVar{Name: "REGISTRY_STORAGE", Value: "swift"},
+		corev1.EnvVar{Name: "REGISTRY_STORAGE_SWIFT_CONTAINER", Value: d.Config.Container},
+		corev1.EnvVar{Name: "REGISTRY_STORAGE_SWIFT_AUTHURL", Value: d.Config.AuthURL},
+		corev1.EnvVar{Name: "REGISTRY_STORAGE_SWIFT_DOMAIN", Value: d.Config.Domain},
+		corev1.EnvVar{Name: "REGISTRY_STORAGE_SWIFT_DOMAINID", Value: d.Config.DomainID},
+		corev1.EnvVar{Name: "REGISTRY_STORAGE_SWIFT_TENANT", Value: d.Config.Tenant},
+		corev1.EnvVar{Name: "REGISTRY_STORAGE_SWIFT_TENANTID", Value: d.Config.TenantID},
+		corev1.EnvVar{Name: "REGISTRY_STORAGE_SWIFT_USERNAME", Value: username},
+		corev1.EnvVar{Name: "REGISTRY_STORAGE_SWIFT_PASSWORD", Value: password},
+		corev1.EnvVar{Name: "REGISTRY_STORAGE_SWIFT_REGION", Value: d.Config.RegionName},
```

Comment:
Can't pass credentials in the clear (security issue)

Reply:
Done